tables/array.py

```diff
@@ -767,21 +798,29 @@ def _read_selection(self, selection, reorder, shape):
             nparr = nparr[tuple(k)].copy()
         return nparr
 
-    def _write_slice(self, startl, stopl, stepl, shape, nparr):
+    def _write_slice(self,
+                     startl: np.ndarray,
+                     stopl: np.ndarray,
+                     stepl: np.ndarray,
+                     shape: list[int],
+                     nparr: np.ndarray) -> None:
         """Write `nparr` in a slice based on `startl`, `stopl` and `stepl`."""
 
         nparr = self._check_shape(nparr, tuple(shape))
         countl = ((stopl - startl - 1) // stepl) + 1
         self._g_write_slice(startl, stepl, countl, nparr)
 
-    def _write_coords(self, coords, nparr):
+    def _write_coords(self, coords: np.ndarray, nparr: np.ndarray) -> None:
         """Write `nparr` values in points defined by `coords` coordinates."""
 
         if len(coords) > 0:
             nparr = self._check_shape(nparr, (len(coords),))
             self._g_write_coords(coords, nparr)
 
-    def _write_selection(self, selection, reorder, shape, nparr):
+    def _write_selection(self,
+                         selection: list[tuple[int, int, int, int, str]],
+                         reorder: Optional[tuple[int, np.ndarray]],
```

Comment:
Probably in this case `ArrayLike` could be used, not sure by the way.
What do you think?

Reply:
I am not exactly sure if this is applicable here as no `Union` type hint is used, thus there is no collapsing.

- For `nparr` the annotation `np.ndarray` is needed as in the following `self._check_shape(nparr, tuple(shape))` call, the `shape` attribute of the numpy array is needed.
- The same applies to `shape`. It needs to be a tuple of integers as we are combining the shape to the `nparr.shape` value in the `self._check_shape` call.
- `reorder` is either `None` or a tuple with two values that do get unpacked. An `ArrayLike` object may just be an array with a single values (or with hundreds of values) and unpacking into two objects will not be possible.
- Similar, each `selection` element consists of four integer and one string (which will be used in `Array.perform_selection` in `tables/hdf5extension.pyx`down the line).

Did you perhaps commented on the wrong line?

Reply:
Sorry, my question was related to the possibility or not to use 
```python
reorder: Optional[tuple[int, ArrayLike]],
```

instead of
```python
reorder: Optional[tuple[int, np.ndarray]],
```

Reply:
Makes sense. Yes, changed it to `ArrayLike`. Thanks!